test/unit/grpcServer/handlers/core/updateStateHandlerFactory.js

```diff
@@ -0,0 +1,168 @@
+const { expect, use } = require('chai');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+const dirtyChai = require('dirty-chai');
+const chaiAsPromised = require('chai-as-promised');
+
+const cbor = require('cbor');
+
+const {
+  UpdateStateTransitionResponse,
+} = require('@dashevo/dapi-grpc');
+
+const getStPacketFixture = require('../../../../../lib/test/fixtures/getStPacketFixture');
+const getStHeaderFixture = require('../../../../../lib/test/fixtures/getStHeaderFixture');
+const GrpcCallMock = require('../../../../../lib/test/mock/GrpcCallMock');
+
+const updateStateHandlerFactory = require(
+  '../../../../../lib/grpcServer/handlers/core/updateStateHandlerFactory',
+);
+const InvalidArgumentGrpcError = require('../../../../../lib/grpcServer/error/InvalidArgumentGrpcError');
+const InternalGrpcError = require('../../../../../lib/grpcServer/error/InternalGrpcError');
+
+use(sinonChai);
+use(chaiAsPromised);
+use(dirtyChai);
+
+describe('updateStateHandlerFactory', () => {
+  let call;
+  let rpcClientMock;
+  let updateStateHandler;
+  let response;
+  let stHeader;
+  let stPacket;
+
+  beforeEach(function beforeEach() {
+    if (!this.sinon) {
+      this.sinon = sinon.createSandbox();
+    } else {
+      this.sinon.restore();
+    }
+
+    const stPacketFixture = getStPacketFixture();
```

Comment:
We don't need to send real ST header and packet here because we don't need to validate request here - just pass it as is.